File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    n = len(keymap)
    key = dict()
    for i in range(n):
        for j in range(len(keymap[i])):
            if key.get(keymap[i][j]) is not None and j < key[keymap[i][j]]:
                key[keymap[i][j]] = j + 1
            elif key.get(keymap[i][j]) is None:
                key[keymap[i][j]] = (j + 1)
    print(key)
    for target in targets:
        total = 0
        for ch in target:
            if key.get(ch) is None:
                total = -1
                break
            else:
                total += key[ch]
        answer.append(total)
    return answer

File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        
        int[] answer = new int[targets.length];
        HashMap<String, Integer> map = new HashMap<>();
        for(String key : keymap){
            String[] str = key.split("");
            for(int i=0 ; i<str.length ; i++){
                if(!map.containsKey(str[i])) map.put(str[i], i+1);
                else if(map.get(str[i]) > i + 1) map.put(str[i], i+1);
            }
        }
        
        for(int i=0 ; i<targets.length ; i++){
            String[] str = targets[i].split("");
            int total = 0;
            for(String ch : str){
                if(!map.containsKey(ch)){
                    total = -1;
                    break;
                }
                total += map.get(ch);
            }
            answer[i] = total;
        }
        return answer;
    }
}